symplyphysics/core/experimental/points/__init__.py

```diff
@@ -0,0 +1,129 @@
+"""
+Points
+======
+
+A **point** in a 3D space is described by 3 coordinates which depend on the choice of the
+coordinate system.
+"""
+
+from __future__ import annotations
+
+from typing import Sequence, Iterator, Any
+
+from sympy import Expr
+
+from ...expr_comparisons import expr_equals
+from ..coordinate_systems import BaseCoordinateSystem
+
+
+class Point:
+    """
+    Class defining the `Point` functionality.
+    """
+
+    _coordinates: tuple[Expr, Expr, Expr]
+    _system: BaseCoordinateSystem
+
+    @property
+    def coordinates(self) -> tuple[Expr, Expr, Expr]:
+        """
+        Coordinates of the point.
+        """
+
+        return self._coordinates
+
+    @property
+    def system(self) -> BaseCoordinateSystem:
+        """
+        Coordinate system the point is attached to.
+        """
+
+        return self._system
+
+    def __init__(
+        self,
+        coordinates: Sequence[Expr],
+        system: BaseCoordinateSystem,
+    ) -> None:
+        if len(coordinates) != 3:
```

Comment:
The issue is that this approach would only make sense for rectilinear (e.g. Cartesian) coordinates, in all other cases we should raise an error (e.g. setting `r` but neither `theta` nor `phi` in a spherical system). I guess we can check if the provided `system` is instance of `RectilinearCoordinateSystem` but I wanted to avoid the dependency on the class hierarchy in the `coordinate_system` module. And I wrote in the comment above about being explicit.

Reply:
Setting zero theta and phi is not forbidden. Why should we raise an error?